openage/convert/value_object/init/game_version.py

```diff
@@ -22,75 +22,92 @@ class Support(enum.Enum):
     breaks = "presence breaks conversion"
 
 
-@enum.unique
-class GameExpansion(enum.Enum):
+class GameBase:
     """
-    An optional expansion to a GameEdition.
+    Common base class for GameEdition and GameExpansion
     """
 
-    AFRI_KING = ("African Kingdoms (HD)",
-                 Support.nope,
-                 {GameFileVersion("resources/_common/dat/empires2_x2_p1.dat",
-                                  {"f2bf8b128b4bdac36ee36fafe139baf1": "1.0c"})},
-                 {
-                     MediaType.GRAPHICS: ["resources/_common/slp/"],
-                     MediaType.SOUNDS: ["resources/_common/sound/"],
-                     MediaType.INTERFACE: ["resources/_common/drs/interface/"],
-                     MediaType.TERRAIN: ["resources/_common/terrain/"]
-                 },
-                 ["aoe2-ak", "aoe2-ak-graphics"])
+    def __init__(self, game_id, **flags):
+        """
+        :param game_id: Unique id for the given game.
+        :type_param: str
+        :param flags: Anything else specific to this version which is useful
+                      for the converter.
+        """
+        self.game_id = game_id
+        self.flags = flags
+
+        self.support_status = None
+        self.game_file_versions = []
+        self.media_paths = {}
+        self.target_modpacks = None
+        self.expansions = []
```

Comment:
Game edition/expansion objects should be immutable (i.e. they should not be altered during conversion for obvious reasons) and therefore, all members should be set during object initialization. You can still use the functions you implemented, but they should be private and only be called from `__init__()`. This means you also have to pass it all necessary values as parameters.

In our architecture view, game editions/expansions are [value objects](https://github.com/SFTtech/openage/blob/master/doc/code/converter/architecture_overview.md#value-object).